common/djangoapps/student/views.py

```diff
@@ -2718,6 +2718,77 @@ def validate_new_email(user, new_email):
         raise ValueError(_('An account with this e-mail already exists.'))
 
 
+def validate_social_link(social_field, new_social_link):
+    """
+    Given a new social link for a user, ensure that the link takes one of the
+    following forms:
+
+    1) A valid url that comes from the correct social site
+    2) A valid username
+    3) A blank value
+    """
+    # Check if the link contains a valid url to the core domain
+    required_url_stub = settings.SOCIAL_PLATFORMS[social_field]['url_stub']
+    contains_social_url = required_url_stub in new_social_link \
+                          and '.' not in new_social_link.replace(required_url_stub, '')
+
+    # Check if the link contains a valid username
+    contains_valid_username = is_valid_username(new_social_link)
+
+    # Check if the link is blank
+    is_empty = not new_social_link
+
+    # Ensure that the new link has either the correct website url, a valid username or is empty.
+    if not (contains_social_url or contains_valid_username or is_empty):
+        raise ValueError(_(
+            ' Make sure that you are providing a valid username or a URL that contains "' +
+            required_url_stub + '". To remove the link from your edX profile, leave this field blank.'
+        ))
+
+
+def format_social_link(social_field, new_social_link):
+    """
+    Given a valid social link for a user, return an absolute url that can be
+    called directly from within a template. The url can take three forms
+    depending on what type of social link was provided:
+
+    1) Given a blank string, return a blank value, representing no link
+    2) Given a valid username, return 'https://www.[platform_name_base][username]'
+    3) Given a valid URL, return 'https://www.[URL]'
+    """
+    # Blank social links are already formatted correctly
+    if not new_social_link:
+        return new_social_link
+
+    url_stub = settings.SOCIAL_PLATFORMS[social_field]['url_stub']
```

Comment:
Here I envision simple code very similar to above, except instead of `validate_social_link()` you will want `get_username_from_social_link()` which either returns the social_link itself or the username stripped from the url.

Then you will have a single `return 'https://www.{}{}'.format(url_stub, username)`. You can thus ignore the original url, and whether or not it had www, or any other issues.  You are just pulling the username and going with it built off the defined url_stub.

Reply:
Pushing this now